scripts/report-open-files.py

```diff
@@ -0,0 +1,120 @@
+#
+# report-open-files.py
+#
+# uftrace-option: --nest-libcall -T open@filter,arg1/s,arg2/e:uft_open_flag,retval/i -T openat@filter,arg1,arg2/s,arg3/e:uft_open_flag,retval/i -T fopen@filter,arg1/s,arg2/s,retval
+#
+
+from __future__ import print_function
+import os
+
+UFTRACE_FUNC = [ "open", "openat", "fopen" ]
+
+UFT_OPEN_MODE_MASK = 0x3
+uft_open_mode = [ "O_RDONLY", "O_WRONLY", "O_RDWR", "O_XXX" ]
+
+uft_open_flag = {
+        "O_CREAT": 0100,
+        "O_EXCL": 0200,
+        "O_NOCTTY": 0400,
+        "O_TRUNC": 01000,
+        "O_APPEND": 02000,
+        "O_NONBLOCK": 04000,
+        "O_DSYNC": 010000,
+        "O_ASYNC": 020000,
+        "O_DIRECT": 040000,
+        "O_LARGEFILE": 0100000,
+        "O_DIRECTORY": 0200000,
+        "O_NOATIME": 01000000,
+        "O_CLOEXEC": 02000000,
+        "O_SYNC": 04010000,
+        "O_PATH": 010000000,
+}
+
+open_map = {}
+openat_map = {}
+fopen_map = {}
+open_pathname = ""
+
+def uftrace_begin():
+    pass
+
+def uftrace_entry(ctx):
+    global open_pathname
+    if ctx.has_key("args"):
+        open_pathname = ctx["args"][0]
+        if ctx["name"] == "open":
+            open_mode = ctx["args"][1]
+            open_map[open_pathname] = open_mode
+        elif ctx["name"] == "openat":
+            open_mode = ctx["args"][2]
+            openat_map[open_pathname] = open_mode
+        elif ctx["name"] == "fopen":
+            open_mode = ctx["args"][1]
+            fopen_map[open_pathname] = open_mode
+
+def uftrace_exit(ctx):
+    global open_pathname
+    if ctx.has_key("retval"):
+        if ctx["name"] == "open":
+            #if ctx["retval"] == -1:
+            if ctx["retval"] == 18446744073709551615:
```

Comment:
This is because python deals with big numbers more than 64 bit.  It's hex: `0xffffffffffffffff`.  I don't know what's the best way to handle this.

Reply:
how about this?

```python
minus_one = (1 << 64) - 1
...

   if ctx["retval"] == minus_one:
```